crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -878,7 +878,7 @@ impl<'db> TypeInferenceBuilder<'db> {
                 ty.inner_type()
             });
         if !bound_ty.is_assignable_to(self.db(), declared_ty) {
-            report_invalid_assignment(&self.context, node, declared_ty, bound_ty);
+            report_invalid_assignment(&self.context, node, declared_ty, bound_ty, None);
```

Comment:
I tried for a while to extract anything useful from `binding.kind(db)`, i.e. the `DefinitionKind` enum. But it didn't change any existing diagnostic messages. And all instances of "Object of type … is not assignable to …" messages I went through would be cases of "… is not assignable to *variable of type* …", which I'm not sure adds a lot of value.